core/pool_vector.cpp

```diff
@@ -69,3 +70,25 @@ void MemoryPool::cleanup() {
 	ERR_EXPLAINC("There are still MemoryPool allocs in use at exit!");
 	ERR_FAIL_COND(allocs_used > 0);
 }
+
+template<>
+String PoolVector<String>::join(String delimiter)
+{
+	int s = size();
+	if (s <= 0)
+		return "";
+
+	Read r = read();
+	bool useDelimiter = !delimiter.empty();
+	StringBuilder sb; sb.reserve_num_appends(useDelimiter ? s+s-1 : s);
```

Comment:
Please don't put two statements on the same line.
Style-wise, there should be spaces in `s + s - 1`.